### PR TITLE
use specified mediasequence for VOD expired sync instead of assuming 0

### PR DESCRIPTION
**Pull request by mjneil**
## Description
Seekable calculations assumed that VOD playlists have an `EXT-X-MEDIA-SEQUENCE` of `0`, however this is not always the case. ~~This fixes the calculations to use the media sequence specified in the playlist~~ This has been updated to now use the `SyncController` to pick a sync point and calculate expired time

### CODE REVIEW

**Comment by gesinger** [comment]
A test we should check for: if a live stream ends, and we switch to a playlist that we haven't seen before (don't have `syncInfo`), or, we tune into a live stream towards its end and rendition switch, then it is possible this may change that behavior.

**Comment by mjneil** [comment]
That scenario is definitely a problem, but it is not a very common case. I think it will be ok to move forward with this and make an issue for that case to address in another PR

File: test/sync-controller.test.js [hunk 1]
```diff
@@ -249,3 +249,151 @@ QUnit.test('Correctly updates time mapping and discontinuity info when probing s
...
+      {
+        duration: 10,
+        uri: '4.ts'
+      }
+    ]
+  };
+
+  let expired = this.syncController.getExpiredTime(playlist, Infinity);
+
+  assert.equal(expired, 100, 'seekable range calculated');
```
Comment (gesinger) [inline test/sync-controller.test.js]: Assertion would probably be clearer using your below format. Something like `estimated expired time using sync info`

File: test/sync-controller.test.js [hunk 2]
```diff
@@ -249,3 +249,151 @@ QUnit.test('Correctly updates time mapping and discontinuity info when probing s
...
+      {
+        duration: 10,
+        uri: '4.ts'
+      }
+    ]
+  };
+
+  expired = this.syncController.getExpiredTime(playlist, Infinity);
+
+  assert.equal(expired, 98.5, 'estimated start time using segmentSync');
```
Comment (gesinger) [inline test/sync-controller.test.js]: Probably better to use `estimated expired time...` instead of start for these and others

**Comment by zhuangs** [comment]
Tested and LGTM

**Pull request merged**